packages/itwinui-react/src/core/Surface/Surface.tsx

```diff
@@ -29,7 +29,47 @@ const getSurfaceElevationValue = (elevation: SurfaceProps['elevation']) => {
   }
 };
 
+// ----------------------------------------------------------------------------
+// Surface.Header component
+
+export type SurfaceHeaderProps = React.ComponentPropsWithRef<'div'>;
+
+const SurfaceHeader = React.forwardRef<HTMLDivElement, SurfaceHeaderProps>(
+  (props, ref) => {
+    const { children, className, ...rest } = props;
+    return (
+      <div className={cx('iui-surface-header', className)} ref={ref} {...rest}>
+        {children}
+      </div>
+    );
+  },
+);
+
+// ----------------------------------------------------------------------------
+// Surface.Body component
+
+export type SurfaceBodyProps = React.ComponentPropsWithRef<'div'>;
+
+const SurfaceBody = React.forwardRef<HTMLDivElement, SurfaceBodyProps>(
+  (props, ref) => {
+    const { children, className, ...rest } = props;
+    return (
+      <div className={cx('iui-surface-body', className)} ref={ref} {...rest}>
+        {children}
+      </div>
+    );
+  },
+);
+
 export type SurfaceProps = {
+  /**
+   * Places children vertically when set to true
+   */
+  useLayout?: boolean;
```

Comment:
I'm also thinking maybe we could remove this prop as it may be possible to infer automatically.

- in CSS we can use [`:has`](https://developer.mozilla.org/en-US/docs/Web/CSS/:has) to detect the presence of child classes
- when `:has` is not supported, we can use react context to update parent state from child

Reply:
Name has been changed to `hasLayout`. Still working on removing this prop